src/query/interpreter.hpp

```diff
@@ -281,7 +243,34 @@ class Interpreter final {
 
   void SetUser(std::string_view username);
 
+  struct SystemTransaction {
+    explicit SystemTransaction(std::unique_lock<utils::ResourceLock> guard, dbms::DbmsHandler &dbms_handler)
+        : system_guard_(std::move(guard)), dbms_handler_{&dbms_handler} {
+      dbms_handler_->NewSystemTransaction();
```

Comment:
Strange that the:
- system lock is `interpreter_context_->system_lock`
- Held in guard that SystemTransaction holds, owned by Interpreter
- dbms_handler controls the actual timestamp values.

Is it possible to system_lock upto the dbms_handler (it will be more inline with where it will need to be for future replication groups)

Reply:
Moved to DBMS